File: Second Seminar/SecondSeminar/src/main/java/com/example/SecondSeminar/member/dto/response/MemberGetResponse.java
package com.example.SecondSeminar.member.dto.response;

import com.example.SecondSeminar.member.domain.Member;
import com.example.SecondSeminar.member.domain.SOPT;

public record MemberGetResponse(
        String name,
        String nickname,
        int age,
        SOPT soptInfo
) {
    public static MemberGetResponse of(Member member) {
        return new MemberGetResponse(member.getName(), member.getNickname(), member.getAge(), member.getSopt());
    }
}

File: Second Seminar/SecondSeminarKotlin/src/main/kotlin/com/example/SecondSeminarKotlin/presentation/dto/response/MemberGetResponse.kt
package com.example.SecondSeminarKotlin.presentation.dto.response

import com.example.SecondSeminarKotlin.domain.Member
import com.example.SecondSeminarKotlin.domain.SOPT

class MemberGetResponse private constructor(
    val name: String,
    val nickname: String,
    val age: Int,
    val soptInfo: SOPT,
) {

    companion object {
        fun of(member: Member): MemberGetResponse {
            return MemberGetResponse(member.name, member.nickname, member.age, member.sopt)
        }
    }
}
